spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -295,4 +296,19 @@ public static List<Expression> partitionMapToExpression(
   public static String toColumnName(NamedReference ref) {
     return DOT.join(ref.fieldNames());
   }
+
+  /**
+   * fetch latest snapshot based on the branch tip
```

Comment:
should add in javadoc about why we want to do thisi nstead of always use `table.snapshot(branch);`